File: app/src/main/java/com/gitsearch/injection/component/AppComponent.java
package com.gitsearch.injection.component;

import com.gitsearch.GitsearchApplication;
import com.gitsearch.injection.module.ActivityBindingModule;
import com.gitsearch.injection.module.ApplicationModule;
import com.gitsearch.injection.module.DataModule;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

import javax.inject.Singleton;

@Singleton
@Component(
        modules = {
                AndroidInjectionModule.class,
                ApplicationModule.class,
                DataModule.class,
                ActivityBindingModule.class
        })
public interface AppComponent extends AndroidInjector<GitsearchApplication> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<GitsearchApplication> {

    }

}

File: app/src/main/java/com/gitsearch/injection/component/AppComponent.kt
package com.gitsearch.injection.component

import com.gitsearch.GitsearchApplication
import com.gitsearch.injection.module.*
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule

import javax.inject.Singleton

@Singleton
@Component(
        modules = [
            AndroidSupportInjectionModule::class,
            ApplicationModule::class,
            AndroidModule::class,
            DataModule::class,
            ActivityBindingModule::class
        ]
)
interface AppComponent : AndroidInjector<GitsearchApplication> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<GitsearchApplication>()

}
